providers/implementations/kdfs/pbkdf2.c

```diff
@@ -327,6 +348,10 @@ static int pbkdf2_derive(const char *pass, size_t passlen,
     }
 
     if (lower_bound_checks) {
+        if (passlen < KDF_PBKDF2_MIN_PASSWORD_LEN) {
```

Comment:
This change is needed

Reply:
Only if an enforced minimum length is required.  The IG does not mandate such.

Reply:
Bouncy Castle FIPS does a check for at least 14 bytes (112 bits).

Reply:
It is possible specify a minimum and maximum password length when requesting CAVP vectors.  Bouncy Castle seems to have understood what to ask for.

My confidence is impared by this comment is in one of the commits in the PR:

> We are choosing a minimum password length of 8 bytes, because NIST's ACVP
> testing uses passwords as short as 8 bytes, and requiring longer passwords
> combined with an implicit indicator (i.e., returning an error) would cause
> the module to fail ACVP testing.

More so, when it is realised that **_we_** have previously requested PBKDF2 testing with password lengths from 1 to 16 bytes thus:
```json
{
	"algorithm": "PBKDF",
	"revision": "1.0",
	"prereqVals": [
		{
			"algorithm": "HMAC",
			"valValue": "same"
		}
	],
	"capabilities": [
		{
			"hmacAlg": [
				"SHA-1",
				"SHA2-224",
				"SHA2-256",
				"SHA2-384",
				"SHA2-512",
				"SHA2-512/256",
				"SHA2-512/224"
			],
			"iterationCount": [
				{
					"min": 1,
					"max": 10000,
					"increment": 1
				}
			],
			"keyLen": [
				{
					"min": 112,
					"max": 4096,
					"increment": 8
				}
			],
			"passwordLen": [
				{
					"min": 8,
					"max": 128,
					"increment": 8
				}
			],
			"saltLen": [
				{
					"min": 128,
					"max": 4096,
					"increment": 8
				}
			]
		}
	]
},
```

I'm in favour of a minimum length, however we'll have to deal with legacy passwords that are shorter.  That means we have to support crazily short lengths since there is no distinction between _new_ and _old_ here.  We could (should?) base this on a configuration parameter of course.

BTW: _14 bytes isn't 112 bits unless all 256 characters are permitted.  Passwords are generally from a _far_ smaller selection._

Reply:
I think it is silly to mandate minimum length in the FIPS module. Of course an application can do it but unless FIPS requirements strictly mandate this, I would not bother.

It just means people will use something like an 8 times repeated character. And no, I do not think we should depend on https://github.com/libpwquality/libpwquality to enforce somewhat stronger passwords :grin:

Reply:
I wasn't aware that we can ask for specific password lengths from the ACVP server. Our lab sits between our testing and the ACVP server, so we typically don't deal directly with the various available options. I'm guessing the information that this possibility exists was just lost in communication.

The FIPS 140-3 IG section D.N does specify that a minimum length must be chosen: "the vendor shall document in the module’s Security Policy the length of a password/passphrase used in key derivation and establish an upper bound for the probability of having this parameter guessed at random." I'm not sure whether it's possible to just select 1 as minimum and document the high probability of guessing the passphrase at random, or whether NIST would end up objecting to that.

Reply:
I think a length of 1 with suitable documentation would meet the requirement as written.  Yes, it a stupid requirement.